File: Leetcode/3174. Clear Digits.cpp
class Solution {
public:
    string clearDigits(string s) {
        stack<char> st;
        for(char ch:s){
            if(ch<'a'){
                if(!st.empty()){
                    st.pop();
                }
            }else{
                st.push(ch);
            }
        }
        string res="";
        while(!st.empty()){
            res+=st.top();
            st.pop();
        }
        reverse(res.begin(),res.end());
        return res;
    }
};

File: Leetcode/3174. Clear Digits.py
class Solution:
    def clearDigits(self, s: str) -> str:
        st=deque()
        for ch in s:
            if ch<'a':
                if len(st)!=0:
                    st.pop()
            else:
                st.append(ch)
        res=""
        while st:
            res+=st[0]
            st.popleft()
        return res
        
